cdap-api/src/main/java/co/cask/cdap/api/dataset/DatasetProperties.java

```diff
@@ -48,6 +48,13 @@ public static Builder builder() {
     return new Builder();
   }
 
+  /**
+   * Helper method to create a DatasetProperties from a map of key/values.
+   */
+  public static DatasetProperties of(Map<String, String> props) {
+    return new DatasetProperties(new HashMap<>(props), null);
```

Comment:
Maybe better to use the `Builder` (`return builder().addAll(props).build()`) to create the `DatasetProperties` so that if there is any creation logic they will all be in one place (e.g. current builder pass in an immutable map to the constructor).